### PR TITLE
DAOS-4409 build: upgrade dpdk to v19.11

### PR DESCRIPTION
**Pull request by tanabarr**
To align with SPDK release v20.01.1.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: dpdk.spec [hunk 1]
```diff
@@ -36,9 +36,6 @@ Source502: set_config.sh
 #Source505: ppc_64-power8-linuxapp-gcc-config
 Source506: x86_64-native-linuxapp-gcc-config
 
-# Patches only in dpdk package
-Patch1: v19.02...67b915b09.patch
-
```
Comment (brianjmurrell) [inline dpdk.spec:41]: You need to provide the patch (https://github.com/spdk/dpdk/compare/v19.11...3fcb1dd395a870c64450f151a53338034b832906 for spdk 20.01.1) that the spdk team adds to dpdk which is basically these commits:

```
* 3fcb1dd39 - (HEAD) contigmem: cleanup properly when load fails (2 weeks ago) <Jim Harris>
* fdb511332 - config: disable build of NXP PFE PMD Driver on ARM (2 months ago) <Alexey Marchuk>
* a0c8f2041 - config: allow the shared library to be built. (3 months ago) <Seth Howell>
* 33918d7f2 - config: comment out vhost (3 months ago) <Darek Stojaczyk>
* 92c570e36 - config: comment out options that we want to override in the makefile (3 months ago) <paul luse>
* 5d0be1f14 - config: don't compile libs we don't use (3 months ago) <Dariusz Stojaczyk>
```

File: dpdk.spec [hunk 2]
```diff
@@ -34,10 +34,7 @@ Source502: set_config.sh
 # @TODO: this causes a warning - fix it?
 #Source504: arm64-armv8a-linuxapp-gcc-config
 #Source505: ppc_64-power8-linuxapp-gcc-config
-Source506: x86_64-native-linuxapp-gcc-config
-
-# Patches only in dpdk package
-Patch1: v19.02...67b915b09.patch
+#Source506: x86_64-native-linuxapp-gcc-config
```
Comment (brianjmurrell) [inline dpdk.spec]: Why are you commenting this out?

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 NAME    := dpdk
 SRC_EXT := xz
 SOURCE   = http://fast.dpdk.org/rel/$(NAME)-$(VERSION).tar.$(SRC_EXT)
```
Comment (brianjmurrell) [inline Makefile]: You can actually remove this line.  It's not needed with the updated `packaging/`.

File: dpdk.spec [hunk 4]
```diff
@@ -81,6 +81,8 @@ ExclusiveArch: x86_64 aarch64 ppc64le
 %define incdir  %{_includedir}/%{name}
 %define pmddir %{_libdir}/%{name}-pmds
 
+%undefine _missing_build_ids_terminate_build
```
Comment (brianjmurrell) [inline dpdk.spec]: Why is this necessary.  It's less-than-ideal to have to add it.

File: x86_64-native-linuxapp-gcc-config [hunk 5]
```diff
@@ -1,11 +1,19 @@
-# -*- cfg-sha: b5f82cd54947d45ceb377032450e026dc687f1cb43eed94c0aa2dd5893af261c
+# -*- cfg-sha: 4bfdb5adfc58968af6739d7098aa8a5ea16ae273a6a45571b926cf71826fd4c0
```
Comment (brianjmurrell) [inline x86_64-native-linuxapp-gcc-config:1]: I don't really know what all of these changes mean, but how did you arrive at them?

File: dpdk.spec (hunk 4 above)
Comment (tanabarr) [reply dpdk.spec]: some of the pmd build IDs are not generated and that terminates to build

File: x86_64-native-linuxapp-gcc-config (hunk 5 above)
Comment (tanabarr) [reply x86_64-native-linuxapp-gcc-config:1]: I extracted the generated config from the SRPM artifact from the build

File: dpdk.spec (hunk 4 above)
Comment (brianjmurrell) [reply dpdk.spec]: Do we know why they are not generated and why this is new with this release?  We really want the build-ids for being able to process core dumps.

**Review by brianjmurrell — APPROVED** [review]
